qt_ui/windows/mission/flight/waypoints/QFlightWaypointList.py

```diff
@@ -112,18 +118,34 @@ def on_changed(self) -> None:
             name = self.model.item(i, 0).text()
             self.flight.flight_plan.waypoints[i].pretty_name = name
 
-    def tot_text(self, flight: Flight, waypoint: FlightWaypoint) -> str:
+    def tot_text(
+        self,
+        flight: Flight,
+        waypoint: FlightWaypoint,
+    ) -> str:
         if waypoint.waypoint_type == FlightWaypointType.TAKEOFF:
+            self.update_last_tot(flight.flight_plan.takeoff_time())
+            self._last_waypoint = waypoint
             return self.takeoff_text(flight)
         prefix = ""
         time = flight.flight_plan.tot_for_waypoint(waypoint)
         if time is None:
             prefix = "Depart "
             time = flight.flight_plan.depart_time_for_waypoint(waypoint)
         if time is None:
-            return ""
+            prefix = ""
+            timedelta = flight.flight_plan.travel_time_between_waypoints(
+                self._last_waypoint, waypoint
```

Comment:
The only concern I have is that `self._last_waypoint` may be uninitialized, though theoretically I can't think of a case where this would actually present itself... 🤔